stylix/autoload.nix

```diff
@@ -8,7 +8,45 @@ builtins.concatLists (
     path: kind:
     let
       file = "${inputs.self}/modules/${path}/${for}.nix";
+      module = import file;
+
+      # Detect whether the file's value has an argument named `mkTarget`
+      useMkTarget =
+        builtins.isFunction module && (builtins.functionArgs module) ? mkTarget;
+
+      # NOTE: `mkTarget` cannot be distributed normally through the module system
+      # due to issues of infinite recursion.
+      mkTarget = import ./mk-target.nix;
     in
-    lib.optional (kind == "directory" && builtins.pathExists file) file
+    lib.optional (kind == "directory" && builtins.pathExists file) (
+      if useMkTarget then
+        { config, ... }@args:
+        let
+          # Based on `lib.modules.applyModuleArgs`
+          #
+          # Apply `mkTarget` as a special arg without actually using `specialArgs`,
+          # which cannot be defined from within a configuration.
+          context =
+            name: ''while evaluating the module argument `${name}' in "${toString file}":'';
+          extraArgs = lib.pipe module [
+            builtins.functionArgs
+            (lib.flip builtins.removeAttrs [ "mkTarget" ])
+            (builtins.mapAttrs (
+              name: _:
+              builtins.addErrorContext (context name) (
+                args.${name} or config._module.args.${name}
+              )
+            ))
+          ];
+          module = module (args // extraArgs // { inherit mkTarget; });
```

Comment:
This is inf-recursion (`module = module` within a recursive scope).

Need to either choose another name for one of the bindings, or inline this expression into the `imports` list below.